launcher/ui/processlist_unix.cpp

```diff
@@ -117,46 +117,65 @@ static ProcDataList unixProcessListPS(const ProcDataList &previous)
     return rc;
 }
 
-static ProcData procIdToProcData(const QString &procId)
+struct ProcIdToProcData
 {
-    ProcData proc;
-    if (!isUnixProcessId(procId))
-        return proc;
+    ProcIdToProcData(const ProcDataList &prev)
+        : previous(prev)
+    {}
 
-    QString filename = QStringLiteral("/proc/");
-    filename += procId;
-    filename += QLatin1String("/stat");
-    QFile file(filename);
-    if (!file.open(QIODevice::ReadOnly))
-        return proc;     // process may have exited
+    typedef ProcData result_type;
 
-    const QStringList data = QString::fromLocal8Bit(file.readAll()).split(' ');
+    ProcData operator()(const QString &procId) const
+    {
+        ProcData proc;
+        if (!isUnixProcessId(procId))
+            return proc;
+
+        QString filename = QStringLiteral("/proc/");
```

Comment:
while you are at it, also change filename = QLatin1String("/proc/") + procId + QLatin1String("/stat") like you do at 155

Reply:
done.